ScrubBar/ScrubBar.swift

```diff
@@ -1,38 +1,62 @@
 import UIKit
 
+public protocol ScrubBarDelegate: class {
+
+    func scrubBar(_ scrubBar: ScrubBar, didSelectItemAt selectedIndex: Int)
+
+}
+
 public class ScrubBar: UIControl {
 
     // MARK: Initialization
 
-    public init(images: [UIImage]) {
-        imageViews = images.map { UIImageView(image: $0) }
+    public init?(items: [ScrubBarItem]) {
+        guard items.count > 0 else { return nil }
+
+        self.items = items
+        imageViews = items.map { UIImageView(image: $0.image) }
         super.init(frame: .zero)
+
         configureImageViews()
         configureStackView()
         configureSelectionView()
     }
 
+    public let items: [ScrubBarItem]
+
     let imageViews: [UIImageView]
 
     private func configureImageViews() {
-        imageViews.forEach {
-            $0.contentMode = .center
-            $0.isAccessibilityElement = true
-            $0.accessibilityTraits = UIAccessibilityTraitButton
+        imageViews.enumerated().forEach { (index, element) in
+            element.contentMode = .center
+            element.isAccessibilityElement = true
+            element.accessibilityTraits = index == selectedIndex ? UIAccessibilityTraitButton | UIAccessibilityTraitSelected : UIAccessibilityTraitButton
+            element.accessibilityLabel = items[index].accessibilityLabel
+            element.tintColor = itemTintColor
         }
+        selectedImageView.tintColor = selectedItemTintColor
     }
 
     required public init?(coder: NSCoder) {
         return nil
     }
 
-    public var selectedSegment = 0 {
+    public var selectedIndex = 0 {
         didSet {
+            guard selectedIndex != oldValue else { return }
+
+            delegate?.scrubBar(self, didSelectItemAt: selectedIndex)
```

Comment:
Consider calling `delegate` when the object is back in consistent state, i.e. end of this property observer.